public/app/features/dashboard/state/settings/actions.ts

```diff
@@ -0,0 +1,31 @@
+import { ThunkResult } from '../../../../types';
+import { setVariableSettings } from './reducer';
+import { DashboardVariablesSettings } from './types';
+import { DashboardModel } from '../DashboardModel';
+import { reportInteraction } from '../../../../../../packages/grafana-runtime';
+
+export function initDashboardSettings(dashboard: DashboardModel): ThunkResult<void> {
+  return function (dispatch) {
+    const variableSettings: DashboardVariablesSettings | undefined = dashboard.getVariableSettings();
```

Comment:
Nit: Does this need to be explicitly typed?

Reply:
Not really, good spot!